test/e2e/framework/ginkgowrapper.go

```diff
@@ -67,6 +125,301 @@ func AnnotatedLocationWithOffset(annotation string, offset int) types.CodeLocati
 
 // ConformanceIt is wrapper function for ginkgo It.  Adds "[Conformance]" tag and makes static analysis easier.
 func ConformanceIt(text string, args ...interface{}) bool {
-	args = append(args, ginkgo.Offset(1))
-	return ginkgo.It(text+" [Conformance]", args...)
+	args = append(args, ginkgo.Offset(1), WithConformance())
+	return It(text, args...)
+}
+
+// It is a wrapper around [ginkgo.It] which supports framework With* labels as
+// optional arguments in addition to those already supported by ginkgo itself,
+// like [ginkgo.Label] and [gingko.Offset].
+//
+// Text and arguments may be mixed. The final text is a concatenation
+// of the text arguments and special tags from the With functions.
+func It(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.It, text, args)
+}
+
+// It is a shorthand for the corresponding package function.
+func (f *Framework) It(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.It, text, args)
+}
+
+// Describe is a wrapper around [ginkgo.Describe] which supports framework
+// With* labels as optional arguments in addition to those already supported by
+// ginkgo itself, like [ginkgo.Label] and [gingko.Offset].
+//
+// Text and arguments may be mixed. The final text is a concatenation
+// of the text arguments and special tags from the With functions.
+func Describe(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.Describe, text, args)
+}
+
+// Describe is a shorthand for the corresponding package function.
+func (f *Framework) Describe(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.Describe, text, args)
+}
+
+// Context is a wrapper around [ginkgo.Context] which supports framework With*
+// labels as optional arguments in addition to those already supported by
+// ginkgo itself, like [ginkgo.Label] and [gingko.Offset].
+//
+// Text and arguments may be mixed. The final text is a concatenation
+// of the text arguments and special tags from the With functions.
+func Context(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.Context, text, args)
+}
+
+// Context is a shorthand for the corresponding package function.
+func (f *Framework) Context(text string, args ...interface{}) bool {
+	return registerInSuite(ginkgo.Context, text, args)
+}
+
+// registerInSuite is the common implementation of all wrapper functions. It
+// expects to be called through one intermediate wrapper.
+func registerInSuite(ginkgoCall func(text string, args ...interface{}) bool, text string, args []interface{}) bool {
```

Comment:
should we have an unit test to "protect" us of future changes here that may break the format?

Reply:
Unit tests for `registerInSuite` certainly can be added. It's complex enough that this is worthwhile. I can add them once we agree on the concept.

Reply:
I added unit tests for all new code.